packages/html/src/datepicker/datepicker.spec.tsx

```diff
@@ -0,0 +1,117 @@
+import { classNames, States, Size, Roundness, FillMode } from '../utils';
+import {
+    Input,
+    InputPrefix,
+    InputSuffix,
+    InputInnerInput,
+    InputClearValue,
+    InputLoadingIcon,
+    InputValidationIcon
+} from '../input';
+import { Button } from '../button';
+import { Popup } from '../popup';
+import { Calendar } from '../calendar';
+
+export const DATEPICKER_CLASSNAME = `k-datepicker`;
+
+const states = [
+    States.hover,
+    States.focus,
+    States.valid,
+    States.invalid,
+    States.required,
+    States.disabled,
+    States.loading
+];
+
+const options = {
+    size: [ Size.small, Size.medium, Size.large ],
+    rounded: [ Roundness.small, Roundness.medium, Roundness.large, Roundness.full ],
+    fillMode: [ FillMode.solid, FillMode.flat, FillMode.outline ],
+};
+
+export type KendoDatePickerOptions = {
+    size?: (typeof options.size)[number] | null;
+    rounded?: (typeof options.rounded)[number] | null;
+    fillMode?: (typeof options.fillMode)[number] | null;
+};
+
+export type KendoDatePickerProps = KendoDatePickerOptions & {
+    prefix?: JSX.Element;
+    suffix?: JSX.Element;
+    value?: string;
+    placeholder?: string;
+    opened?: boolean;
+};
+
+export type KendoDatePickerState = { [K in (typeof states)[number]]?: boolean };
+
+export const DatePicker = (
+    props: KendoDatePickerProps &
+        KendoDatePickerState &
+        React.HTMLAttributes<HTMLSpanElement>
+) => {
+    const {
+        prefix,
+        suffix,
+        value,
+        placeholder,
+        size,
+        rounded,
+        fillMode,
+        hover,
+        focus,
+        valid,
+        invalid,
+        required,
+        loading,
+        disabled,
+        opened,
+        ...other
+    } = props;
+
+
+    return (
+        <>
```

Comment:
The `DatePicker` does not use the `Picker` component, while the DropDownList and other dropdowns are using it. I'm not really sure which one would be correct, but can we please double-check that?

Reply:
the DatePicker is an input with a button, while the `Picker` component is part of these components which use span with classes `k-picker k-picker-solid`, etc. It may seem confusing, but we are basing the input/picker component names on the classes used for styling